web/src/client/questions.js

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) [2022] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+const DBUS_CONFIG = {
+  questions: {
+    path: "/org/opensuse/DInstaller/Questions1",
+    ifaces: {
+      objectManager: "org.freedesktop.DBus.ObjectManager"
+    }
+  },
+  question: {
+    ifaces: {
+      question: "org.opensuse.DInstaller.Question1",
+      luksActivation: "org.opensuse.DInstaller.Question.LuksActivation1"
+    }
+  }
+};
+
+const QUESTION_TYPES = {
+  generic: "generic",
+  luksActivation: "luksActivation"
+};
+
+/**
+ * Builds a question from the given D-Bus question
+ *
+ * @param {Object} dbusQuestion
+ * @returns {Object}
+*/
+function buildQuestion(dbusQuestion) {
+  const getIfacesAndProperties = (dbusQuestion) => Object.values(dbusQuestion)[0];
```

Comment:
I am not sure if I would define the function here. If you were using a closure over `dbusQuestion` it would be fine, but that's not the case (you are passing it anyway as an argument). I am not saying that it is wrong, it is just it simply looks strange to me.

Reply:
That's is why I commented before that I'd move those methods outside. Now that there are two opinions in the same side... lets move them.